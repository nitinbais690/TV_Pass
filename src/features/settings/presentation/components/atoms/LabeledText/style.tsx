import { StyleSheet } from 'react-native';
import { scale } from 'qp-common-ui';
import { appFonts, appFontStyle } from 'core/styles/AppStyles';

export function labeledTextStyle(appColors: any) {
    return StyleSheet.create({
        label: {
            color: appColors.secondary,
            ...appFontStyle.body3,
            fontFamily: appFonts.semibold,
            opacity: 0.3,
        },
        title: {
            paddingTop: scale(4, 0),
            color: appColors.secondary,
            ...appFontStyle.body3,
            fontFamily: appFonts.semibold,
        },
    });
}
