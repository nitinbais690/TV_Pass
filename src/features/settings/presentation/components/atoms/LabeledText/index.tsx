import React from 'react';
import { View, Text } from 'react-native';
import { labeledTextStyle } from './style';
import useAppColors from 'core/presentation/hooks/use-app-colors';

const LabeledText = (props: LabeledTextProps) => {
    const appColors = useAppColors();
    const style = labeledTextStyle(appColors);

    return (
        <View style={props.style}>
            <Text style={style.label}>{props.label}</Text>
            <Text style={style.title}>{props.text}</Text>
        </View>
    );
};

interface LabeledTextProps {
    label: string;
    text: string;
    style?: {};
}

export default LabeledText;
