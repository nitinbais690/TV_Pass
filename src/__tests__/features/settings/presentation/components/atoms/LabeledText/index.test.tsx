import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import { StyleSheet } from 'react-native';
import LabeledText from 'features/settings/presentation/components/atoms/LabeledText';

jest.mock('core/presentation/hooks/use-app-colors');

describe('labeledText', () => {
    const label = 'Email';
    const text = 'kunal@mailinator.com';

    it('renders correctly', () => {
        const { container, getByText } = render(<LabeledText label={label} text={text} />);
        expect(getByText(label)).toBeDefined();
        expect(getByText(text)).toBeDefined();
        expect(toJSON(container)).toMatchSnapshot();
    });

    it('render with style', () => {
        const style = StyleSheet.create({
            label: {
                paddingTop: 15,
            },
        });

        const { container, getByText } = render(<LabeledText label={label} text={text} style={style.label} />);
        expect(getByText(label)).toBeDefined();
        expect(getByText(text)).toBeDefined();
        expect(toJSON(container)).toMatchSnapshot();
    });
});
