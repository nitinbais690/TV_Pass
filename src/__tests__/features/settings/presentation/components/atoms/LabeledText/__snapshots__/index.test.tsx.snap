// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`labeledText render with style 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "paddingTop": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "ProximaNova-Semibold",
          "fontSize": 18,
          "fontWeight": "600",
          "opacity": 0.3,
        }
      }
    >
      Email
    </Text>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "ProximaNova-Semibold",
          "fontSize": 18,
          "fontWeight": "600",
          "paddingTop": 4,
        }
      }
    >
      kunal@mailinator.com
    </Text>
  </View>
</View>
`;

exports[`labeledText renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "ProximaNova-Semibold",
          "fontSize": 18,
          "fontWeight": "600",
          "opacity": 0.3,
        }
      }
    >
      Email
    </Text>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "ProximaNova-Semibold",
          "fontSize": 18,
          "fontWeight": "600",
          "paddingTop": 4,
        }
      }
    >
      kunal@mailinator.com
    </Text>
  </View>
</View>
`;
